dv/uvm/core_ibex/riscv_dv_extension/ibex_log_to_trace_csv.py

```diff
@@ -131,29 +132,44 @@ def convert_operands_to_abi(operand_str):
     return ",".join(operand_list)
 
 
-def process_trace(trace):
-    """ Process instruction trace """
-    process_imm(trace)
-    if trace.instr == 'jalr':
-        n = ADDR_RE.search(trace.operand)
-        if n:
-            trace.imm = get_imm_hex_val(n.group("imm"))
+def expand_trace_entry(trace, operands):
+    '''Expands a CSV trace entry for a single instruction.
+
+    Operands are added to the CSV entry, converting from the raw
+    register naming scheme (x0, x1, etc...) to ABI naming (a1, s1, etc...).
+
+    '''
+    operands = process_imm(trace.instr, trace.pc, operands)
+    trace.instr, operands = \
+        convert_pseudo_instr(trace.instr, operands, trace.binary)
+
+    # process any instructions of the form:
+    # <instr> <reg> <imm>(<reg>)
+    n = ADDR_RE.search(operands)
+    if n:
+        trace.imm = get_imm_hex_val(n.group("imm"))
+        operands = ','.join([n.group("rd"), n.group("rs1"), n.group("imm")])
+
+    # Convert the operands into ABI format for the function coverage flow,
+    # and store them into the CSV trace entry.
+    trace.operand = convert_operands_to_abi(operands)
 
 
-def process_imm(trace):
+def process_imm(instr_name, pc, operands):
     """Process imm to follow RISC-V standard convention"""
-    if trace.instr in ['beq', 'bne', 'blt', 'bge', 'bltu', 'bgeu', 'c.beqz',
-                       'c.bnez', 'beqz', 'bnez', 'bgez', 'bltz', 'blez',
-                       'bgtz', 'c.j', 'j', 'c.jal', 'jal']:
-        idx = trace.operand.rfind(',')
+    if instr_name in ['beq', 'bne', 'blt', 'bge', 'bltu', 'bgeu', 'c.beqz',
+                      'c.bnez', 'beqz', 'bnez', 'bgez', 'bltz', 'blez',
+                      'bgtz', 'c.j', 'j', 'c.jal', 'jal']:
+        idx = operands.rfind(',')
         if idx == -1:
-            imm = trace.operand
-            imm = str(sint_to_hex(int(imm, 16) - int(trace.pc, 16)))
-            trace.operand = imm
-        else:
-            imm = trace.operand[idx + 1:]
-            imm = str(sint_to_hex(int(imm, 16) - int(trace.pc, 16)))
-            trace.operand = trace.operand[0:idx + 1] + imm
+            imm = operands
+            return str(sint_to_hex(int(imm, 16) - int(pc, 16)))
+
+        imm = operands[idx + 1:]
+        imm = str(sint_to_hex(int(imm, 16) - int(pc, 16)))
+        return operands[0:idx + 1] + imm
+    else:
+        return operands
```

Comment:
since we now rely on the return value from `process_imm`, we need to make sure we can pass through the operands to prevent not recognizing any instruction that isn't a branch or jump.

Reply:
Oops! I didn't notice that, thanks!

Some linters complain about
```python
if some_test:
    ...
    return 123
else:
   ...
```
because the "else" isn't strictly necessary (if you go in to the first branch, you always return). You've currently got an example of this at line 171. That complaint is why I suggested dropping the `else` after the `idx == -1` check. I don't have strong feelings here, but it seems some linter has slapped me across the knuckles enough times to do it automatically :-)

For this particular example, I'd actually suggest switching the outer `if` around. So you get:
```python
if instr_name not in [some, long, list]:
   return operands

idx = operands.rfind(',')
...
```
This avoids the single line floating off away from the check.

Anyway, this is a very nitty style question: I'm happy either way!

Reply:
Sure, I have no issues changing this slightly!